src/scraper.py

```diff
@@ -0,0 +1,187 @@
+    """
+    Read a static html (inspection_page.html).
+    Return the bytes content and the encoding.
+    Used for testing.
+    """
+    import io
+    files = io.open(path, 'rb')
+    response = files.read()
+    files.close()
+    encoding = 'utf-8'
+    return response, encoding
+
+
+def parse_source(html, encoding='utf-8'):
+    """
+    Parse the rsponse body using BeautifulSoup.
+    Return the parsed object for further processing.
+    """
+    parsed = BeautifulSoup(html, 'html5lib', from_encoding=encoding)
+    return parsed
+
+
+def extract_data_listings(html):
+    """
+    Takes in the parsed HTML. Return a list of the restaurant listing
+    container nodes.
+    """
+    id_finder = re.compile(r'PR[\d]+~')
+    return html.find_all('div', id=id_finder)
+
+
+def has_two_tds(element):
+    """
+    Takes an element as an argument.
+    Return 'True' if the element is both a <tr> and contains
+    two <td> elements within it, 'False' otherwise.
+    """
+    elem_is_tr = element.name == 'tr'
+    elem_has_td_children = element.find_all('td', recursive=False)
+    has_two_td = len(elem_has_td_children) == 2
+    return elem_is_tr and has_two_td
+
+
+def clean_data(element):
+    """
+    Takes a cell as an argument. Return the tag.string attribute
+    with extraneous characters stripped.
+    """
+    data = element.string
+    try:
+        return data.strip(" \n:-")
+    except AttributeError:
+        return u""
+
+
+def extract_restaurant_metadata(listing):
+    """
+    Take in the listing for a single restaurant.
+    Return a Python dict with metadata:
+    {<label>: <value>}.
+    """
+    listing_dict = {}
+    metadata_rows = listing.find('tbody').find_all(
+        has_two_tds, recursive=False
+    )
+    for row in metadata_rows:
+        for td in row.find_all('td', recursive=False):
+            key_row, val_row = row.find_all('td', recursive=False)
+            label_present = clean_data(key_row)
+            if label_present:
+                label_dict = label_present
+            else:
+                label_dict = ''
+            listing_dict.setdefault(label_dict, []).append(clean_data(val_row))
+    return listing_dict
+
+
+def is_inspection_row(element):
+    """A filter function. Take in an HTML element.
+    Return 'True' if the element matches the criteria:
+    1 - is a <tr> element,
+    2 - contains 4 <td> elements,
+    3 - word 'inspection' is in the text of it's first <td>,
+    4 - word 'inspection' is not the 1st word in that <td>.
+    Return 'False' otherwise.
+    """
+    is_tr = element.name == 'tr'
+    if not is_tr:
+        return False
+    td_children = element.find_all('td', recursive=False)
+    has_four_td = len(td_children) == 4
+    first_td_text = clean_data(td_children[0]).lower()
+    contains_word = 'inspection' in first_td_text
+    does_not_start = not first_td_text.startswith('inspection')
+    return is_tr and has_four_td and contains_word and does_not_start
```

Comment:
This is a great use case for the builtin function `all()`, which takes an iterable (e.g. list or tuple) returns `True` if all statements in the iterable evaluate `True`.